src/components/TransitionStaggered/src/TransitionStaggered.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+	<transition-group
+		v-bind="$attrs"
+		tag="div"
+		@enter="handleEnter"
+		@leave="handleLeave"
+	>
+		<slot
+			:data-load-index="loadIndex"
+		/>
+	</transition-group>
+</template>
+
+<script>
+import { staggeredFloatUpFn, floatDownFn } from '@square/maker/utils/transitions';
+
+const defaultItemCount = [
+	{
+		minWidth: 0,
+		itemCount: 1,
+	},
+];
+
+export default {
+	props: {
+		// Number of items to stagger at a time for a viewport
+		// E.g. [
+		// 	{
+		// 		minWidth: 0,
+		// 		staggerItemCount: 1,
+		// 	},
+		// 	{
+		// 		minWidth: 840,
+		// 		staggerItemCount: 4,
+		// 	},
+		// ]
+		staggerItemCount: {
```

Comment:
What about a name like `itemsPerRow` for this prop? Still doesn't capture the media query support, but more clear about what the number means.

Reply:
There's something about the name bugging me too -- also the complexity of the data object we're passing in. I'm wondering if we can split it up into multiple props, but I'm at a loss right now on what it should be. 🤔

Reply:
I'm thinking about how you would construct a grid like this with flex or css grid -- perhaps we can model the props the same way you'd do with regular CSS.  Rather than doing calculations based on the viewport width and passing data about the width the consumer should just consider the grid they want and pass those details.

For that matter, now I'm kind of wondering if a list or collection component responsible for the layout of that type of content should be used and the transition is more baked in with the skeleton loaders.

So many ways to skin this cat...

Reply:
I tried an earlier version that had a `skeleton` slot and more layout handling. But it would have required refactoring of how BX views handle loading states and grids/grid props currently. It felt like a task for a second pass once we can cover all current use cases.

For now, in order to have a Maker integration that can be used for testing on Website, the simplest component I needed was a transition that handles a staggered load animation for a group of elements. It would still be useful within a more comprehensive loading component.

As far as checking the window viewport - that's also how Maker handles applying different transition animations for Modals on mobile and web. For now the transition component has no awareness of how a list of elements is displayed, it just needs to know how to group them for animation sequencing.